internal/operands/node-labeller/resources.go

```diff
@@ -0,0 +1,295 @@
+package node_labeller
+
+import (
+	"kubevirt.io/ssp-operator/internal/common"
+
+	secv1 "github.com/openshift/api/security/v1"
+	apps "k8s.io/api/apps/v1"
+	core "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Resources from node-labeller
+const (
+	ServiceAccountName = "kubevirt-node-labeller"
+	DaemonSetName      = "kubevirt-node-labeller"
+	ConfigMapName      = "kubevirt-cpu-plugin-configmap"
+
+	nfdVolumeName            = "nfd-source"
+	nfdVolumeMountPath       = "/etc/kubernetes/node-feature-discovery/source.d/"
+	configMapVolumeName      = "cpu-config"
+	configMapVolumeMountPath = "/config"
+)
+
+var commonLabels = map[string]string{
+	"app": "kubevirt-node-labeller",
+}
+
+var cpuPluginConfigmap = `
+    obsoleteCPUs:
+      - "486"
+      - "pentium"
+      - "pentium2"
+      - "pentium3"
+      - "pentiumpro"
+      - "coreduo"
+      - "n270"
+      - "core2duo"
+      - "Conroe"
+      - "athlon"
+      - "phenom"
+    minCPU: "Penryn"
+`
+
+var configMapData = map[string]string{
+	"cpu-plugin-configmap.yaml": cpuPluginConfigmap,
+}
+
+func ClusterRoleName(namespace string) string {
+	const clusterRoleName = "kubevirt-node-labeller"
+	return clusterRoleName + "-" + namespace
+}
+
+func ClusterRoleBindingName(namespace string) string {
+	return ServiceAccountName + "-" + namespace
+}
+
+func newClusterRole(namespace string) *rbac.ClusterRole {
+	return &rbac.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   ClusterRoleName(namespace),
+			Labels: commonLabels,
+		},
+		Rules: []rbac.PolicyRule{{
+			APIGroups: []string{""},
+			Resources: []string{"nodes"},
+			Verbs:     []string{"get", "update", "patch"},
+		}},
+	}
+}
+
+func newServiceAccount(namespace string) *core.ServiceAccount {
+	return &core.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ServiceAccountName,
+			Namespace: namespace,
+		},
+	}
+}
+
+func newClusterRoleBinding(namespace string) *rbac.ClusterRoleBinding {
+	return &rbac.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: ClusterRoleBindingName(namespace),
+		},
+		RoleRef: rbac.RoleRef{
+			Kind:     "ClusterRole",
+			Name:     ClusterRoleName(namespace),
+			APIGroup: "rbac.authorization.k8s.io",
+		},
+		Subjects: []rbac.Subject{{
+			Kind:      "ServiceAccount",
+			Name:      ServiceAccountName,
+			Namespace: namespace,
+		}},
+	}
+}
+
+func newConfigMap(namespace string) *core.ConfigMap {
+	return &core.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ConfigMapName,
+			Namespace: namespace,
+		},
+		Data: configMapData,
+	}
+}
+
+func buildContainer(name string, image string, args []string, cmd []string) *core.Container {
+	return &core.Container{
+		Name:  name,
+		Image: image,
+		//VolumeMounts: volMount,
+		Command: cmd,
+		Args:    args,
+	}
+}
+
+func volumeMounts(name []string, path []string) []core.VolumeMount {
```

Comment:
This function is mostly called with arrays with single elements. It would be clearer to create the `core.VolumeMount` struct directly inline, instead of calling this function.

Reply:
Done. Deleted the function and added the values inline.